tests/acceptance/request_test.py

```diff
@@ -99,6 +114,7 @@ def test_400_if_required_body_is_missing(test_app):
     ).status_code == 400
 
 
+@pytest.mark.xfail
```

Comment:
Another option is to move the code that validates `json_body` before the code that validates the other parameters. This seems bad because of the  dependency on checking the `json_body` before `body`, but on the other hand, it seems like reading one is mutually exclusive with reading the other. 

I'm hesitant to say we can trust the content-type header. Isn't the whole point of this test as it is originally written that we can not trust it?